tests/infrastructure/crds.go

```diff
@@ -47,15 +46,16 @@ var _ = DescribeInfra("CRDs", func() {
 
 	It("[test_id:5177]Should have structural schema", func() {
 		ourCRDs := []string{crds.VIRTUALMACHINE, crds.VIRTUALMACHINEINSTANCE, crds.VIRTUALMACHINEINSTANCEPRESET,
-			crds.VIRTUALMACHINEINSTANCEREPLICASET, crds.VIRTUALMACHINEINSTANCEMIGRATION, crds.KUBEVIRT,
+			crds.VIRTUALMACHINEINSTANCEREPLICASET, crds.VIRTUALMACHINEPOOL,
+			crds.VIRTUALMACHINEINSTANCEMIGRATION, crds.MIGRATIONPOLICY,
+			crds.KUBEVIRT,
 			crds.VIRTUALMACHINESNAPSHOT, crds.VIRTUALMACHINESNAPSHOTCONTENT,
+			crds.VIRTUALMACHINECLONE,
+			crds.VIRTUALMACHINEEXPORT,
```

Comment:
We should think about something that enforces us to align this test if a CRD is added.

Reply:
+1 Any idea?

Reply:
Still thinking about it :thinking:

Reply:
One easy thing could be smth like this.
In `pkg/virt-operator/resource/generate/components/crds.go`:
```go
var (
	virtualmachine                   = "virtualmachines"
	virtualmachineinstance           = "virtualmachineinstances"
	virtualmachineinstancepreset     = "virtualmachineinstancepresets"
	virtualmachineinstancereplicaset = "virtualmachineinstancereplicasets"
	virtualmachineinstancemigration  = "virtualmachineinstancemigrations"
	kubevirt                         = "kubevirts"
	virtualmachinepool               = "virtualmachinepools"
	virtualmachinesnapshot           = "virtualmachinesnapshots"
	virtualmachinesnapshotcontent    = "virtualmachinesnapshotcontents"
	virtualmachineexport             = "virtualmachineexports"
	migrationpolicy                  = "migrationpolicies"
	virtualmachineclone              = "virtualmachineclones"
	PreserveUnknownFieldsFalse       = false
)

var CRDs = map[string]string{
	virtualmachine:                   "virtualmachines." + virtv1.VirtualMachineInstanceGroupVersionKind.Group,
	virtualmachineinstance:           "virtualmachineinstances." + virtv1.VirtualMachineInstanceGroupVersionKind.Group,
	virtualmachineinstancepreset:     "virtualmachineinstancepresets." + virtv1.VirtualMachineInstancePresetGroupVersionKind.Group,
	virtualmachineinstancereplicaset: "virtualmachineinstancereplicasets." + virtv1.VirtualMachineInstanceReplicaSetGroupVersionKind.Group,
	virtualmachineinstancemigration:  "virtualmachineinstancemigrations." + virtv1.VirtualMachineInstanceMigrationGroupVersionKind.Group,
	kubevirt:                         "kubevirts." + virtv1.KubeVirtGroupVersionKind.Group,
	virtualmachinepool:               "virtualmachinepools." + poolv1.SchemeGroupVersion.Group,
	virtualmachinesnapshot:           "virtualmachinesnapshots." + snapshotv1beta1.SchemeGroupVersion.Group,
	virtualmachinesnapshotcontent:    "virtualmachinesnapshotcontents." + snapshotv1beta1.SchemeGroupVersion.Group,
	virtualmachineexport:             "virtualmachineexports." + exportv1beta1.SchemeGroupVersion.Group,
	migrationpolicy:                  "migrationpolicies." + migrationsv1.MigrationPolicyKind.Group,
	virtualmachineclone:              "virtualmachineclones." + clonev1alpha1.VirtualMachineCloneKind.Group,
}
```

and then in this test do:
```go
		for _, name := range crds.CRDs {
			ext, err := extclient.NewForConfig(virtClient.Config())
			Expect(err).ToNot(HaveOccurred())

			crd, err := ext.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), name, metav1.GetOptions{})
			Expect(err).ToNot(HaveOccurred())

			Expect(crd).To(matcher.HaveConditionMissingOrFalse(v1ext.NonStructuralSchema))
		}
```
But probably it's ugly

Reply:
@fossedihelm I'll add my idea as well, listing resources (from API as you can do with kubectl) and filtering `kubevirt.io`. Then we can check the crds.

Reply:
Sounds good, but compare with what? :)

Reply:
/approve
/hold
Holding in case you want to tackle the above in this PR. Feel free to unhold.

Reply:
/hold cancel